src/span.ts

```diff
@@ -1,12 +1,14 @@
 import * as noop from './noop';
+import SpanContext from './span_context';
+import Tracer from './tracer';
 
 /**
  * Span represents a logical unit of work as part of a broader Trace. Examples
  * of span might include remote procedure calls or a in-process function calls
  * to sub-components. A Trace has a single, top-level "root" Span that in turn
  * may have zero or more child Spans, which in turn may have children.
  */
-export default class Span {
+export class Span {
```

Comment:
Well, as a TS user, I can only use this as soon as the implementation of my choice adopts it. So I wouldn't say it doesn't matter, but I would rather have a cleaner API.

But the outer API _would_ change. You would have to export the noop implementation under a different name if e.g. `Tracer` is now just an interface (look at my example above).

Noop isn't something separate right now really, all the classes are documented of implementing noop behavior, and I already used them that way